RELEASES.md

```diff
@@ -80,8 +80,6 @@ For patch releases, only the version on the existing major and minor version bra
 
 * A new [tag](https://github.com/elastic/apm-server/releases) will automatically be created on GitHub.
 
-* Bump the version in anticipation of the next release, e.g. [after 7.5.1 release](https://github.com/elastic/apm-server/pull/3045/files) bump to 7.5.2. Prepare this PR ahead of time, but only merge after release when pinged by the release manager.
```

Comment:
[**For reviewers**] CMIIW, I don't see a need for this anymore. This entails that the release process doesn't require **any changes** to be done on the release or main branch post-release date.